components/list/demo/list-item-sample.js

```diff
@@ -52,34 +80,37 @@ class ListItemSample extends ListItemMixin(ListItemCheckboxMixin(LitElement)) {
 		const classes = { hovering: this._hovering, focusing: this._focusing };
 
 		return html`
-			<d2l-list-item-placement-marker></d2l-list-item-placement-marker>
-			<d2l-list-item-generic-layout ?grid-active="${this.role === 'rowgroup'}">
-				${ this.draggable ? html`
-				<div slot="outside-control">${ this._renderDragHandle ? this._renderDragHandle() : '=' }</div>
-				` : nothing }
-				${this.selectable ? html`
-				<div slot="control">${ this._renderCheckbox() }</div>
-				<div slot="control-action">${ this._renderCheckboxAction(null, this._contentId) }</div>
-				` : nothing }
-				${ this.href ? html`
-				<a slot="content-action"
-					href="${this.href}"
-					aria-labelledby="${this._contentId}"
-					@mouseenter="${this._handleMouseEnter}"
-					@mouseleave="${this._handleMouseLeave}"
-					@focus="${this._handleFocus}"
-					@blur="${this._handleBlur}"></a>
-				` : nothing }
-				<div slot="content"
-					id="${this._contentId}"
-					class="d2l-list-item-content ${ classMap(classes) }">
-					<slot></slot>
-				</div>
-
-				<div slot="actions">
-					<slot name="actions"></slot>
-				</div>
-			</d2l-list-item-generic-layout>
+			${this._renderTopPlacementMarker(html`<d2l-list-item-placement-marker></d2l-list-item-placement-marker>`)}
+			${this._renderDropArea()}
+			<div class="d2l-list-item-drag-image">
+				<d2l-list-item-generic-layout ?grid-active="${this.role === 'rowgroup'}">
+					${this._renderDragHandle((dragHandle) => html`<div slot="outside-control">${dragHandle}</div>`)}
+					${this._renderDraggableArea((dragArea) => html`<div slot="outside-control-action">${dragArea}</div>`)}
+					${this.selectable ? html`
+					<div slot="control">${ this._renderCheckbox() }</div>
+					<div slot="control-action">${ this._renderCheckboxAction(null, this._contentId) }</div>
+					` : nothing }
```

Comment:
There is a difference in pattern here. `${this._renderDraggableArea((dragArea) => html`<div slot="outside-control-action">${dragArea}</div>`)}` is more complicated but because it is more powerful and the drag n drop mixin takes use of this power. We probably will want to switch the checkbox mixin to do the same.

Reply:
This results in new functions being created for each `render` - in the context of event handlers, the linter will catch this for perf reasons.  The downside to moving it to functions on the class though, is that it's less clear to readers what's going to get rendered.

Reply:
I'll split it off into a method and hopefully the name can hopefully make it still easier to read. The docs on how to make a new list item style should write them in the more readable way and make a note to split it off.